### DISCUSSION TITLE
how to pass arguments to validators

### DISCUSSION THREAD

**Discussion by debdutgoswami** [pydantic/pydantic#2775] [Question] [answered]
I have a pydantic model on which I need to do some validation with enums. I have written a global function like,

```python3
def enum_validator(v: str, EnumClass: Enum):
    if isinstance(v, EnumClass):
        return v
    return EnumClass(v)
```

And I am calling the validator as,

```python3
_first_order_enum = validator("first_order", EnumClass=YesNo, allow_reuse=True)(enum_validator)
```

It gives me an error 

```python3
TypeError: validator() got an unexpected keyword argument 'EnumClass'
```

Now, my question is how do I pass the Enum Class to the defined function?

**Comment by PrettyWood** [comment, marked as answer]
Hi @debdutgoswami 
You probably want something like
```py
from functools import partial
_first_order_enum = validator("first_order", allow_reuse=True)(partial(enum_validator, EnumClass=YesNo))
```

**Reply by debdutgoswami** [reply]
Yes this is exactly what I ended up doing. Thanks.

**Reply by jdaok** [reply]
Does this still work for you? I keep getting attribute error: functools.partial object has no attribute __name__ ..
